formatter/src/formatter/element.rs

```diff
@@ -155,6 +159,14 @@ mod tests {
             })
         }};
     }
+    macro_rules! format_element_with_self_closing_tag {
```

Comment:
This would be a great improvement, but since I'm not good with macros I did not manage to get this to work.
 
With the two arms written in that order, I hit a recursion overflow so I think the macro was just matching over the first arm repeatedly.
With the two arms in the opposite order, I had a parsing error for all uses of the macro.
It feels to me like the tt type is also matching with the expr type, making the macro match the first arm either way (overflowing in the first case, and looking for an expr when none are given in the second).

Can you try to update the macro ? I can update the tests to work with the new macro then

Reply:
Ah I see, I am fine to keep things simple and extract it into a different macro, like so:

```rust
    macro_rules! format_element {
        ($($tt:tt)*) => {{
            format_element_with!(Default::default(), $($tt)*)
        }};
    }

    macro_rules! format_element_with {
        ($settings:expr, $($tt:tt)*) => {{
            let element = element! { $($tt)* };
            format_with(FormatterSettings { max_width: 40, ..$settings }, |formatter| {
                formatter.element(&element)
            })
        }};
    }
```

so you can use it like so:

```rust
    #[test]
    fn self_closing_non_void_element_no_children_self_closing_tag() {
        let settings = FormatterSettings {
            closing_tag_style: ClosingTagStyle::SelfClosing, // <--
            ..Default::default()
        };
        let formatted = format_element_with! { settings, < div / > };
        insta::assert_snapshot!(formatted, @"<div/>");
    }

    #[test]
    fn non_self_closing_non_void_element_no_children_self_closing_tag() {
        let settings = FormatterSettings {
            closing_tag_style: ClosingTagStyle::NonSelfClosing, // <--
            ..Default::default()
        };
        let formatted = format_element_with! { settings, < div / > };
        insta::assert_snapshot!(formatted, @"<div></div>");
    }
```

but considering a fair amount of tests is specific to closing style, you could also define a macro like so:
```rust
    macro_rules! format_element_with_closing_style {
        ($style:expr, $($tt:tt)*) => {{
            let element = element! { $($tt)* };
            format_with(FormatterSettings { closing_tag_style: $style, ..Default::default() }, |formatter| {
                formatter.element(&element)
            })
        }};
    }
    
    
    
    #[test]
    fn non_self_closing_void_element_no_children_separate_closing_tag() {
        let formatted = format_element_with_closing_style! { ClosingTagStyle::NonSelfClosing, < input >< / input > };
        insta::assert_snapshot!(formatted, @"<input />");
    }
```